Implemented configurables to remove upgrades from

TeslaRage opened the pull request:
Closes #26

- New config file `XComUI.ini`
- Removed instances of `eInvSlot_PrimaryWeapon` and read from config instead
- Removed instances of `X2WeaponTemplate` casting and used `ItemState` for all check purpose (to support armor upgrades based on CHL)

TeslaRage committed:
Implemented configurables to remove upgrades from

TeslaRage committed:
Missed out x2proj file

TeslaRage committed:
Loc update

TeslaRage committed:
Fixed bug where:
- upgrades are not removed from items not in HQ

Xymanek commented on PrototypeArmoury/Config/XComUI.ini:
```diff
@@ -0,0 +1,3 @@
+[PrototypeArmoury.UIUtilities_PA]
+
++SlotsToStrip=eInvSlot_PrimaryWeapon
```
Please end files with a single empty line

Xymanek commented on PrototypeArmoury/Src/PrototypeArmoury/Classes/UIUtilities_PA.uc:
```diff
@@ -136,9 +139,9 @@ static function OnStripUpgradesDialogCallback (Name eAction)
 				}
 				
-				if (!ItemState.HasBeenModified() && !WeaponTemplate.bAlwaysUnique)
+				if (!ItemState.HasBeenModified() && !ItemState.GetMyTemplate().bAlwaysUnique)
```
Why not `EquipmentTemplate`?

Xymanek commented on PrototypeArmoury/Src/PrototypeArmoury/Classes/UIUtilities_PA.uc:
```diff
@@ -136,9 +139,9 @@ static function OnStripUpgradesDialogCallback (Name eAction)
+				if (!ItemState.HasBeenModified() && !ItemState.GetMyTemplate().bAlwaysUnique)
 				{
-					if (WeaponTemplate.bInfiniteItem)
+					if (ItemState.GetMyTemplate().bInfiniteItem)
```
Same here

Xymanek commented on PrototypeArmoury/Src/PrototypeArmoury/Classes/UIUtilities_PA.uc:
```diff
@@ -153,21 +156,23 @@ static function OnStripUpgradesDialogCallback (Name eAction)
-					EquippedUpgrades = ItemState.GetMyWeaponUpgradeTemplates();
-
-					if (ShouldRemoveNicknamedUpgrades(ItemState))
+					ItemState = UnitState.GetItemInSlot(SlotToStrip);
```
This assumes that all upgradable items are in single-item slots. For sake of completion, please rework this bit to support upgradable items in utility and other multi-item slots

Xymanek commented on PrototypeArmoury/Src/PrototypeArmoury/Classes/UIUtilities_PA.uc:
```diff
@@ -9,6 +9,8 @@ var localized string strStripUpgradesTooltip;
 var localized string strStripUpgradesConfirm;
 var localized string strStripUpgradesConfirmDesc;
 
+var config array<EInventorySlot> SlotsToStrip;
```
```suggestion
var config array<EInventorySlot> SlotsToStripUpgrades;
```

TeslaRage committed:
Implemented changes based on PR review
- Renamed config variable
- `EquipmentTemplate` instead of `ItemState.GetMyTemplate()`
- Replaced `GetItemInSlot` with `GetAllItemsInSlot`
